test/PrepStmt/client2.cpp

```
@@ -83,7 +83,7 @@ int main() {
 			fprintf(stderr, " mysql_stmt_init(), out of memory\n");
 			exit(EXIT_FAILURE);
 		}
-		sprintf(buff,"SELECT %u + ?",(uint32_t)mt_rand()%NUMPRO);
+		snprintf(buff,sizeof(buff),"SELECT %u + ?",(uint32_t)mt_rand()%NUMPRO);
```

Cppcheck (2.21.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)